### PR TITLE
Add yaml config validator and testing

### PR DESCRIPTION
**Pull request by debermudez**
Add a yaml file validator that checks for underscore usage and against a list of valid options.

### CODE REVIEW

File: tests/test_yaml_options.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        self.assertFalse(
+            TestYamlOptions.yaml_arg_validator.is_valid_option(hyphen_option))
+
+    def test_multiple_options(self):
+        options = {"client_max_retries", "profile_model", "triton-server-flags"}
+        count = 0
+        for entry in options:
+            if TestYamlOptions.yaml_arg_validator.is_valid_option(entry):
+                count += 1
+            self.assertEqual(count, 1,
```
[the remark this answers is not in the document]
Comment (debermudez) [reply tests/test_yaml_options.py:50]: yep, fixed

File: model_analyzer/config/input/config_command.py [hunk 2]
```diff
@@ -96,6 +97,11 @@ def set_config_values(self, args):
         else:
             yaml_config = None
 
+        validator = YamlConfigValidator()
+        if yaml_config:
+            for option in yaml_config.keys():
+                validator.is_valid_option(option)
```
Comment (tgerdesnv) [inline model_analyzer/config/input/config_command.py:103]: this function returns a bool, but you don't do anything with it?
will logger.error actually error out? I think not? Maybe keep a flag (`failed |= validator.is_valid_option()`), and then after: `if failed: raise exception`

File: tests/test_yaml_options.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        self.assertFalse(
+            TestYamlOptions.yaml_arg_validator.is_valid_option(
+                misspelled_option))
+
+    def test_using_hyphens_not_underscores(self):
+        hyphen_option = "triton-server-flags"
+        self.assertFalse(
+            TestYamlOptions.yaml_arg_validator.is_valid_option(hyphen_option))
+
+    def test_multiple_options(self):
```
Comment (tgerdesnv) [inline tests/test_yaml_options.py:41]: let's expand this test a bit (a few more options), and add a comment since it isn't clear why the count is 1

**Review by tgerdesnv — CHANGES_REQUESTED** [review]

File: model_analyzer/config/input/yaml_config_validator.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+class YamlConfigValidator:
+    """
+    Validate all options from the yaml file.
+    """
+
+    _valid_yaml_options = set({})
+    _initialized = False
+
+    @staticmethod
+    def is_yaml_file_valid(yaml_file):
```
[the remark this answers is not in the document]
Comment (debermudez) [reply model_analyzer/config/input/yaml_config_validator.py]: updated method name and added a docstring

File: tests/test_yaml_options.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                        -
+                            kind: KIND_GPU
+                            count: [1,2]                        
+            """)
+        # yapf: enable
+
+        yaml_config = self._load_config_file(yaml_content)
+        self.assertTrue(YamlConfigValidator.is_yaml_file_valid(yaml_config))
+
+    def test_invalid_yaml_file(self):
```
[the remark this answers is not in the document]
Comment (debermudez) [reply tests/test_yaml_options.py:70]: added a comment

File: model_analyzer/config/input/config_command.py (hunk 2 above)
Comment (debermudez) [reply model_analyzer/config/input/config_command.py]: done

File: model_analyzer/config/input/yaml_config_validator.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            valid &= YamlConfigValidator._is_valid_option(option)
+
+        if not valid:
+            raise TritonModelAnalyzerException(
+                "The Yaml configuration file is incorrect. Please check the logged errors for the specific options that are causing the issue."
+            )
+
+    @staticmethod
+    def _is_valid_option(option):
+        YamlConfigValidator._create_valid_option_set()
```
Comment (tgerdesnv) [inline model_analyzer/config/input/yaml_config_validator.py:52]: not needed anymore

File: tests/test_yaml_options.py (hunk 3 above)
Comment (debermudez) [reply tests/test_yaml_options.py:40]: I added 2 more cases: `"DURATION_seconds", ""`  
Added a comment to reflect failing options

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**